components/event-service/internal/events/subscribed/events.go

```diff
@@ -34,84 +33,63 @@ type Event struct {
 }
 
 type eventsClient struct {
-	subscriptionsClient SubscriptionsGetter
-	namespacesClient    NamespacesClient
+	triggerLister kneventinglister.TriggerLister
 }
 
 //NewEventsClient function creates client for retrieving all active events
-func NewEventsClient(subscriptionsClient SubscriptionsGetter, namespacesClient NamespacesClient) EventsClient {
+func NewEventsClient(client versioned.Interface) EventsClient {
+	informerFactory := externalversions.NewSharedInformerFactory(client, 0)
+	ctx := signals.NewContext()
+	go informerFactory.Start(ctx.Done())
```

Comment:
This call is [non-blocking](https://github.com/kubernetes/client-go/blob/v11.0.0/informers/factory.go#L133), so it shouldn't run in a goroutine.

Reply:
Thanks... Changes committed